airflow/models/taskinstance.py

```diff
@@ -1367,6 +1367,11 @@ class TaskInstance(Base, LoggingMixin):
 
     trigger = relationship("Trigger", uselist=False, back_populates="task_instance")
     triggerer_job = association_proxy("trigger", "triggerer_job")
+    queued_by_job = relationship(
+        "Job",
+        back_populates="task_instances_enqueued",
+        primaryjoin="Job.id == foreign(TaskInstance.queued_by_job_id)",
```

Comment:
I think we should have uselist=False since it was removed from backref at the other end?

Reply:
> it was removed from backref at the other end?

TBH it the other way round, references right now setup in the both way. 

> I think we should have uselist=False

Maybe it should be setup `uselist=False` in both way if it one-to-one relationship